x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/authorization/ElasticInferenceServiceAuthorization.java

```diff
@@ -12,16 +12,20 @@
 
 import java.util.EnumSet;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
 import java.util.stream.Collectors;
 
 /**
- * Provides a structure for governing which models (if any) a cluster has access to according to the upstream Elastic Inference Service.
- * @param enabledModels a mapping of model ids to a set of {@link TaskType} to indicate which models are available and for which task types
+ * This is a helper class for managing the response from {@link ElasticInferenceServiceAuthorizationHandler}.
  */
-public record ElasticInferenceServiceAuthorization(Map<String, EnumSet<TaskType>> enabledModels) {
+public class ElasticInferenceServiceAuthorization {
```

Comment:
I refactored this because we need both the authorized task types and the authorized models.